From 44a9803ec6e0d981f7efc8617dff8c0d6a687055 Mon Sep 17 00:00:00 2001
From: Derek Weitzel <djw8605@gmail.com>
Date: Wed, 11 Oct 2023 08:56:37 -0500
Subject: [PATCH] [XrdHttp] Fix max to min when reading chunk length

Small change to using the minimum of either the used buffer size
or the first 13 characters of a chunk to read the chunk length.
---
 src/XrdHttp/XrdHttpReq.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/XrdHttp/XrdHttpReq.cc b/src/XrdHttp/XrdHttpReq.cc
index 78cbcd2c7..2202ae7ef 100644
--- a/src/XrdHttp/XrdHttpReq.cc
+++ b/src/XrdHttp/XrdHttpReq.cc
@@ -1334,7 +1334,7 @@ int XrdHttpReq::ProcessHTTPReq() {
             // This is to prevent a malicious client from sending a very large chunk size
             // or a malformed chunk request.
             // 1TB in base-16 is 0x40000000000, so only allow 11 characters, plus the CRLF
-            long long max_chunk_size_chars = std::max(static_cast<long long>(prot->BuffUsed()), static_cast<long long>(13));
+            long long max_chunk_size_chars = std::min(static_cast<long long>(prot->BuffUsed()), static_cast<long long>(13));
             for (; idx < max_chunk_size_chars; idx++) {
               if (prot->myBuffStart[idx] == '\n') {
                 found_newline = true;
-- 
2.34.1

